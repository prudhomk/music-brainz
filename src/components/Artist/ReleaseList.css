.ReleaseList {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ReleaseList li {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 250px;
  height: auto;
  cursor: pointer;
  list-style: none;
  margin: 5px;
  padding: 5px;
  border: 2px solid black;
  box-shadow: 2px 2px 10px 2px;
  transition: width 0.5s;
  object-fit: contain;
}

.ReleaseList img {
  object-fit: contain;
  height: 200px;
  width: 200px;
}

.ReleaseList li:hover {
  background-color: pink;
  width: 300px;
}