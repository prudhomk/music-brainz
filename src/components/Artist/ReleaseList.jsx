import React, { useEffect, useState } from 'react';
import Release from './Release';
import { Link, useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { mungeReleases, releaseSearch } from '../../services/musicApi';
import styles from './ReleaseList.css';

export default function ReleaseList() {
  const [releases, setReleases] = useState([]);
  const [loading, setLoading] = useState(true);
  const [offSet, setOffSet] = useState(5);
  const [page, setPage] = useState(1);
  const { id } = useParams();


  useEffect(async () => {
    const releases = await releaseSearch(id, offSet);
    const mungedReleases = await  mungeReleases(releases);
    setReleases(mungedReleases);
    setLoading(false);

  }, []);

  const handleIncrement = async () => {
    setPage((prevPage) => prevPage + 1);
    setOffSet((prevOffSet) => prevOffSet + 5);
    const query = await releaseSearch(id, offSet);

    const searchedRelease = await mungeReleases(query);
    setReleases(searchedRelease);
  };

  const handleDecrement = async () => {
    setPage((prevPage) => prevPage - 1);
    setOffSet((prevOffSet) => prevOffSet - 5);
    const query = await releaseSearch(id, offSet);

    const searchedRelease = await mungeReleases(query);
    setReleases(searchedRelease);
  };


  if(releases) {
    const releaseResult = releases.map((release) => (   
      <>
        <li key={release.id}>
          <Link to={`/${release.id}/recordings`}>
            <Release {...release} />
          </Link>
        </li>
      </>
      
    ));

    return (
      <>
        <h1>List of Releases</h1>
        <button 
          disabled={page <= 1}
          onClick={handleDecrement}>
              -
        </button> 
      Page: {page}
        <button onClick={handleIncrement}>
    +
        </button>

        <ul className={styles.ReleaseList}>
          {releaseResult}
        </ul>
      </>
    );
  }
  
  return (
    <div>
        PLACEHOLDER    
    </div>
  );
}

ReleaseList.propTypes = {
  releaseList: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      date: PropTypes.string.isRequired
    })
  )
};
