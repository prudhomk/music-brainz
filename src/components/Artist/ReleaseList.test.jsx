import React from 'react';
import { render } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import ReleaseList from './ReleaseList';
import { MemoryRouter } from 'react-router-dom';

const server = setupServer(
  rest.get(
    'http://musicbrainz.org/ws/2/recording?release=id&fmt=json',
    (req, res, ctx) => {
      return res(
        ctx.json(
          {
            'id': '1',
            'title': 'banana',
            'date': '09-09-09'
          }
        )
      );
    }
  )
);

describe('Main display component snapshot', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());
  
  it('render ReleaseList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ReleaseList />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
