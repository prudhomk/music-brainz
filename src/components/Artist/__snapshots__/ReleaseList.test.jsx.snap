// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main display component snapshot render ReleaseList 1`] = `
<DocumentFragment>
  <h1>
    List of Releases
  </h1>
  <button
    disabled=""
  >
    -
  </button>
  Page: 1
  <button>
    +
  </button>
  <ul
    class="ReleaseList"
  />
</DocumentFragment>
`;
